lib/internal/modules/esm/load.js

```diff
@@ -29,13 +29,12 @@ const {
  * @param {ESModuleContext} context used to decorate error messages
  * @returns {Promise<{ responseURL: string, source: string | BufferView }>}
  */
-async function getSource(url, context) {
+function getSource(url, context) {
   const { protocol, href } = url;
   const responseURL = href;
   let source;
   if (protocol === 'file:') {
-    const { readFile: readFileAsync } = require('internal/fs/promises').exports;
-    source = await readFileAsync(url);
+    source = readFileSync(url);
```

Comment:
This converts to a blocking op, but I'm not sure if that matters.

Reply:
The CommonJS `loadSource` uses `readFileSync`:

https://github.com/nodejs/node/blob/fbe37d501ec45c79ca3904b392f8858e63e42c0b/lib/internal/modules/cjs/loader.js#L1128

though CommonJS is fundamentally sync. I think it should be fine to use the same in ESM; @joyeecheung or anyone else, do you have any concerns?

Reply:
Blocking fs is generally not problematic when your fs is fast enough (and not taking on the async overhead might make it faster, as https://github.com/nodejs/performance/issues/151 shows).